src/crate/crash/command.py

```diff
@@ -494,9 +503,40 @@ def main():
 
     crate_hosts = [host_and_port(h) for h in args.hosts]
     error_trace = args.verbose > 0
+
+    force_passwd_prompt = args.password
+    password = None
+
+    # If password prompt is not forced try to get it from env. variable.
+    if not force_passwd_prompt:
+        password = os.environ.get('CRATEPW', None)
+
+    # Prompt for password immediately to avoid that the first time trying to
+    # connect to the server runs into an `Unauthorized` excpetion
+    # is_tty = False
+    if force_passwd_prompt and not password and is_tty:
+        password = getpass()
+
+    # Tries to create a connection to the server.
```

Comment:
I think this is pretty self explanatory and the comment wouldn't be necessary, but feel free to keep it.